docs/en/02_Developer_Guides/19_GraphQL/02_working_with_dataobjects/04_inheritance.md

```diff
@@ -0,0 +1,178 @@
+---
+title: DataObject inheritance
+summary: Learn how inheritance is handled in DataObject types
+---
+
+# Working with DataObjects
+
+[CHILDREN asList]
+
+## DataObject inheritance
+
+The inheritance pattern used in the ORM is a tricky thing to navigate in a GraphQL API, mostly owing
+to the fact that there is no concept of inheritance in GraphQL types. The main tools we have at our
+disposal are [interfaces](https://graphql.org/learn/schema/#interfaces) and [unions](https://graphql.org/learn/schema/#union-types) to deal with this type of architecture, but in practise, it quickly becomes cumbersome.
+For instance, just adding a subclass to a DataObject can force the return type to change from a simple list
+of types to a union of multiple types, and this would break frontend code.
+
+While more conventional, unions and interfaces introduce more complexity, and given how much we rely
+on inheritance in Silverstripe CMS, particularly with `SiteTree`, inheritance in GraphQL is handled in a less
+conventional but more ergonomic way using a plugin called `inheritance`.
+
+### Introducing pseudo-unions
+
+Let's take a simple example. Imagine we have this design:
+
+```
+> SiteTree
+  > Page
+    > NewsPage
+    > Contact Page
+```
+
+Now, let's expose `Page` to graphql:
+
+```yaml
+SilverStripe\GraphQL\Schema\Schema:
+  schemas:
+    default:
+      models:
+        Page:
+          fields:
+            title: true
+            content: true
+            bannerImage: true
+          operations: '*'
+```
+
+Here's how we can query the inherited fields:
+
+```graphql
+query readSiteTrees {
+  nodes {
+    title
+    content
+    __extends {
+      Page {
+         bannerImage {
+           url
+         }
+      }
+    }
+  }
+}
+```
+
+The `__extends` field is semantically aligned with is PHP counterpart -- it's an object whose fields are the
```

Comment:
There's been a few iterations on this since the docs were written, should be fixed now. I've added a reminder to review the docs on https://github.com/silverstripe/silverstripe-graphql/issues/296